libraries/AP_IRLock/AP_IRLock_I2C.cpp

```diff
@@ -0,0 +1,163 @@
+/*
+   This program is free software: you can redistribute it and/or modify
+   it under the terms of the GNU General Public License as published by
+   the Free Software Foundation, either version 3 of the License, or
+   (at your option) any later version.
+
+   This program is distributed in the hope that it will be useful,
+   but WITHOUT ANY WARRANTY; without even the implied warranty of
+   MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+   GNU General Public License for more details.
+
+   You should have received a copy of the GNU General Public License
+   along with this program.  If not, see <http://www.gnu.org/licenses/>.
+ */
+
+/*
+ * AP_IRLock_I2C.cpp
+ *
+ * Based on AP_IRLock_PX4 by MLandes
+ *
+ * See: http://irlock.com/pages/serial-communication-protocol
+ */
+#include <AP_HAL/AP_HAL.h>
+#include "AP_IRLock_I2C.h"
+#include <stdio.h>
+
+extern const AP_HAL::HAL& hal;
+
+#define IRLOCK_I2C_ADDRESS		0x54
+
+#define IRLOCK_SYNC			0xAA55AA55
+
+void AP_IRLock_I2C::init()
+{
+    dev = std::move(hal.i2c_mgr->get_device(1, IRLOCK_I2C_ADDRESS));
```

Comment:
Why only look at bus 1? Actually weren't this looking at bus 0 before (and eventually 1 for VR Brain boards)?